gdal/ogr/ogrsf_frmts/cad/libopencad/cadgeometry.h

```diff
@@ -0,0 +1,666 @@
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in all
+ *  copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ *  SOFTWARE.
+ *******************************************************************************/
+#ifndef CADGEOMETRIES_H
+#define CADGEOMETRIES_H
+
+#include "cadobjects.h"
+#include "cadcolors.h"
+
+#include <array>
+
+using namespace std;
+
+class CADAttdef;
+
+class CADAttrib;
+
+/**
+ * @brief The Matrix class
+ */
+class Matrix
+{
+public:
+              Matrix();
+    void      translate( const CADVector& vector );
+    void      rotate( double rotation );
+    void      scale( const CADVector& vector );
+    CADVector multiply( const CADVector& vector ) const;
+protected:
+    array<double, 9> matrix;
+};
+
+/**
+ * @brief Base CAD geometry class
+ */
+class CADGeometry
+{
+public:
+    CADGeometry();
+    virtual ~CADGeometry();
+    /**
+     * @brief The CAD geometry types enum
+     */
+    enum GeometryType
+    {
+        UNDEFINED = 0,
+        POINT,
+        CIRCLE,
+        LWPOLYLINE,
+        ELLIPSE,
+        LINE,
+        POLYLINE3D,
+        TEXT,
+        ARC,
+        SPLINE,
+        SOLID,
+        RAY,
+        HATCH, // NOT IMPLEMENTED
+        IMAGE,
+        MTEXT,
+        MLINE,
+        XLINE,
+        FACE3D,
+        POLYLINE_PFACE,
+        ATTRIB,
+        ATTDEF
+    };
+
+    enum GeometryType getType() const;
+    double            getThickness() const;
+    void              setThickness( double thicknes );
+    RGBColor          getColor() const;
+    void              setColor( RGBColor color );// TODO: in 2004+ ACI is not the only way to set the color.
+
+    vector<CADAttrib> getBlockAttributes() const;
+    void              setBlockAttributes( const vector<CADAttrib>& value );
+
+    vector<string> getEED() const;
+    void           setEED( vector<string> eed );
+
+    virtual void print() const                     = 0;
+    virtual void transform( const Matrix& matrix ) = 0;
+protected:
+    vector<CADAttrib> blockAttributes; // attributes of block reference this geometry is attached to.
```

Comment:
attributes -> Attributes and move to the line before.
As before, fit almost all lines to 80 cols